File: CS/DXApplication5/Program.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection;
using System.Threading;
using System.Windows.Forms;
using DevExpress.Data.Filtering;
using DevExpress.LookAndFeel;
using DevExpress.Skins;
using DevExpress.UserSkins;
using DevExpress.XtraEditors;
using DXSample.CustomFunctions;

namespace DXSample {
    static class Program {
        /// <summary>
        /// The main entry point for the application.
        /// </summary>
        [STAThread]
        static void Main() {
            IsWeekendFunction.Register();
            IsDaysInterval.Register();
            NotBeginsWithFunction.Register();
            CriteriaOperator.QueryCustomFunctions += OnQueryCustomUIFunctions;
            UserLookAndFeel.Default.SkinName = "The Bezier";
            Application.EnableVisualStyles();
            Application.SetCompatibleTextRenderingDefault(false);
            Application.Run(new Main());
            
        }
        private static void OnQueryCustomUIFunctions(object sender, DevExpress.Data.Filtering.CustomFunctionEventArgs e) {
            //if(e.PropertyType == typeof(string)) {
            //    e.Add(NotBeginsWithFunction.FunctionName);
            //}
        }
    }
}

File: VB/DXApplication5/Program.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Reflection
Imports System.Threading
Imports System.Windows.Forms
Imports DevExpress.Data.Filtering
Imports DevExpress.LookAndFeel
Imports DevExpress.Skins
Imports DevExpress.UserSkins
Imports DevExpress.XtraEditors
Imports DXSample.CustomFunctions

Namespace DXSample
	Friend Module Program
		''' <summary>
		''' The main entry point for the application.
		''' </summary>
		<STAThread>
		Sub Main()
			IsWeekendFunction.Register()
			IsDaysInterval.Register()
			NotBeginsWithFunction.Register()
			AddHandler CriteriaOperator.QueryCustomFunctions, AddressOf OnQueryCustomUIFunctions
			UserLookAndFeel.Default.SkinName = "The Bezier"
			Application.EnableVisualStyles()
			Application.SetCompatibleTextRenderingDefault(False)
			Application.Run(New Main())

		End Sub
		Private Sub OnQueryCustomUIFunctions(ByVal sender As Object, ByVal e As DevExpress.Data.Filtering.CustomFunctionEventArgs)
			'if(e.PropertyType == typeof(string)) {
			'    e.Add(NotBeginsWithFunction.FunctionName);
			'}
		End Sub
	End Module
End Namespace